app/models/vm_or_template.rb

```diff
@@ -123,7 +123,8 @@ class VmOrTemplate < ActiveRecord::Base
   include ReportableMixin
 
   virtual_column :active,                               :type => :boolean
-  virtual_column :archived,                             :type => :boolean,    :uses => [:host, :storage]
+  virtual_column :archived,                             :type => :boolean,
+                 :sql  => "(CASE WHEN vms.ems_id IS NULL and vms.storage_id IS NULL THEN true ELSE false END)"
```

Comment:
@kbrock You are listing the vms table directly in the SQL, but I wonder if it should do something like `quote_table_name`.  I'm also curious if any joins end up in the SQL for virtual columns, if this could cause any issues.  I can't think of test cases that might exercise that path though.